### PR TITLE
Impelment time_format enhancement to use custom time formats

### LINKED ISSUE #2105: Generic CSV parser: time conversion and time format
When CSV parser parses the file and event line contains time format like %d-%m-%Y, fuzzy time conversion fails to guess correct time format. Example of the feed is http://benkow.cc/export.php
Fuzzy conversion works with such date entries:
>>> DateTime.convert_fuzzy('12-08-2021')
'2021-12-08T00:00:00'
And correct with days >12 :
>>> DateTime.convert_fuzzy('16-08-2021')
'2021-08-16T00:00:00'

Probably better option to use from_format:
>>> DateTime.convert_from_format('12-08-2021', '%d-%m-%Y')
'2021-08-12T00:00:00+00:00'

### PR DESCRIPTION
**Pull request by gutsohnCERT**
Implementation and enhancement of time_format parameter to accept custom date formats (Fix #2105). Usage e.g.  'time_format': 'from_format|%d-%m-%Y %H:%M:%S'

### CODE REVIEW

File: intelmq/tests/bots/parsers/generic/custom_date_format.csv.license [hunk 1]
```diff
@@ -0,0 +1,2 @@
+SPDX-FileCopyrightText: 2022 gutsohnCERT
```
Comment (waldbauer-certat) [inline intelmq/tests/bots/parsers/generic/custom_date_format.csv.license:1]:
```suggestion
SPDX-FileCopyrightText: 2022 CERT.at GmbH
```

File: intelmq/tests/bots/parsers/generic/custom_date_format.csv [hunk 2]
```diff
@@ -0,0 +1,13 @@
+ # comment with space before
+	# comment with tab before
+	  # comment with mix of tabs and spaces before
+ # below line contains spaces only
+   
+ # below line contains tabs only
+			
+ # below line contains mix of tabs and spaces only
+
```
Comment (waldbauer-certat) [inline intelmq/tests/bots/parsers/generic/custom_date_format.csv:9]:
```suggestion
```

We could remove those lines, as they are not part of this test.

**Review by waldbauer-certat — CHANGES_REQUESTED** [review]

File: intelmq/tests/bots/parsers/generic/custom_date_format.csv.license (hunk 1 above)
Comment (aaronkaplan) [reply intelmq/tests/bots/parsers/generic/custom_date_format.csv.license:1]: > 

I wonder why you want to change the license header from gutsohnCERT to CERT.at Gmbh. That does not make sense for me...

Comment (sebix) [reply intelmq/tests/bots/parsers/generic/custom_date_format.csv.license:1]: "gutsohnCERT" was an employee of CERT.at

**Comment by gethvi** [comment]
This is superseded by #2329. Close?

**Pull request closed without merging**